TW-1017: Adding new webhooks tutorial and re-writing blocks list (#4742)
postmanlabs/postman-docs: src/pages/docs/postman-flows/reference/blocks-list.md

---
title: "Postman Flows blocks"
updated: 2023-04-12
---

The blocks available in Postman Flows are described below.

* [Information blocks](#information-blocks)
* [Decision blocks](#decision-blocks)
* [Repeating blocks](#repeating-blocks)
* [Action blocks](#action-blocks)

> * [Output block](#output-block)

Reviewer: ```suggestion
* [Output block](#output-blocks)
```

Reviewer: This was my error. 😅

<img alt="Template block" src="https://assets.postman.com/postman-labs-docs/all-blocks/template-block.png" width="245px"/>

> Allows large amounts of formatted text to be entered. The values in the variables section can be included in the text using `{{key}}` where `key` is the name of your variable.

Reviewer: ```suggestion
Enables large amounts of formatted text to be entered. The values in the variables section can be included in the text using `{{key}}` where `key` is the name of your variable.
```

**Variables** - Named information or assigned values that can be included in the text using the `{{key}}` format.

> **Out** - sends the information entered in the text box to the next block.

Reviewer: ```suggestion
**Out** - Sends the information entered in the text box to the next block.
```

[Example](https://www.postman.com/postman/workspace/flows-snippets/flow/63bc9412ead4e97c45493eb0)

### Date & Time

<img alt="Date &amp; Time block" src="https://assets.postman.com/postman-labs-docs/all-blocks/date-and-time-block.png" width="221px"/>

Outputs the specified date and time.

[Example](https://www.postman.com/postman/workspace/flows-snippets/flow/63bc6e7782ae416b9e6bc10a)

### Date

<img alt="Date block" src="https://assets.postman.com/postman-labs-docs/all-blocks/date-block.png" width="164px"/>

Outputs the specified date.

[Example](https://www.postman.com/postman/workspace/flows-snippets/flow/63bc6e72ead4e97c45493e9a)

<!-- vale Postman.Avoid = NO -->
### Now

<img alt="Now block" src="https://assets.postman.com/postman-labs-docs/all-blocks/now-block.png" width="69px"/>

Outputs the current date and time.

[Example](https://www.postman.com/postman/workspace/flows-snippets/flow/63bc6e61e308d202595085c9)
<!-- vale Postman.Avoid = YES -->

### Select

<img alt="Select block" src="https://assets.postman.com/postman-labs-docs/all-blocks/select-block.png" width="142px"/>

> Selects a specific part of the information output from another block. If the input information is from a send request block with a saved example, a drop-down will pop up that has all the information available to choose from and clicking on that information will auto-fill the select block.

Reviewer: ```suggestion
Selects a specific part of the information output from another block. If the input information is from a send request block with a saved example, a drop-down will pop up that has all the information available to choose from and selecting that information will auto-fill the select block.
```

  **Text entry block** - Freeform query using [FQL](/docs/postman-flows/flows-query-language/introduction-to-fql/) that evalutes to true or false.

  **Data** - Receives information or an event from another block.

  **Outputs** - Sends the information to either the True or False pathway depending on the value of the FQL condition.

  [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/63bcba94f3155f2e86b54eb0)
<!-- vale Postman.Spelling = YES -->

### Evaluate

<img alt="Evaluate block" src="https://assets.postman.com/postman-labs-docs/all-blocks/updated-evaluate-block.png" width="318px"/>

Executes [FQL](/docs/postman-flows/flows-query-language/introduction-to-fql/) to change information and work with variables.

**Variables** - Named information or assigned values that can be accessed using FQL.

**Text entry block** - Freeform query using FQL.

**Out** - Outputs the result of the query.

[Example](https://www.postman.com/postman/workspace/flows-snippets/flow/63bc960882ae416b9e6bc11f)

## Repeating blocks

### Repeat

<img alt="Repeat block" src="https://assets.postman.com/postman-labs-docs/all-blocks/repeat-block.png" width="129px"/>

> Will run all of the blocks that come after it the specified number of times until a **Collect** block is encountered in which case it will stop repeating all blocks that come after the **Collect** block.

Reviewer: ```suggestion
Runs all of the blocks that come after it the specified number of times until a **Collect** block is encountered in which case it will stop repeating all blocks that come after the **Collect** block.
```

**Count** - The number of times to run the block.

> **Start** - starts running the block when it receives input from another block.

Reviewer: ```suggestion
**Start** - Runs the block when it receives input from another block.
```

**Index** - The number of the current run starting at 0 for the first run.

[Example](https://www.postman.com/postman/workspace/flows-snippets/flow/63b898422e67b05a8333e16a)

### For

<img alt="For block" src="https://assets.postman.com/postman-labs-docs/all-blocks/for-block.png" width="111px"/>

> Will run all of the blocks that come after it once for each item in the input list until a **Collect** block is encountered in which case it will stop repeating all blocks that come after the **Collect** block.

Reviewer: ```suggestion
Runs all of the blocks that come after it once for each item in the input list until a **Collect** block is encountered in which case it will stop repeating all blocks that come after the **Collect** block.
```

**List** - A list of values like `[1,2,3]` or `["one","two","three"]`.

**Start** - Receives an input to run the block.

**Item** - Outputs a single item in the list.

[Example](https://www.postman.com/postman/workspace/flows-snippets/flow/63b89820a365885b55bc51f6)

### Collect

<img alt="Collect block" src="https://assets.postman.com/postman-labs-docs/all-blocks/collect-block.png" width="118px"/>

**For** and **Repeat** blocks will stop repeating once this block is reached. Every item will be put into a new list and then output.

**Item** - Takes in a single item from a for or repeat sequence.

> **List** - Creates a list of all items sent to this block in the for or repeat.

Reviewer: ```suggestion
**List** - Creates a list of all items sent to this block in the for or repeat sequence.
```

<img alt="Delay block" src="https://assets.postman.com/postman-labs-docs/all-blocks/delay-block.png" width="212px"/>

Waits the specified amount of time before allowing information through.

**Data (input)** - Starts the delay when it receives input from another block.

**Data (output)** - Once the delay is complete, passes through whatever information was input.

**Delay (in ms)** - The amount of time in milliseconds to wait.

[Example](https://www.postman.com/postman/workspace/flows-snippets/flow/63bcbaf6ead4e97c45493eb9)

### Send Request

<img alt="Send Request block" src="https://assets.postman.com/postman-labs-docs/all-blocks/send-request-block.png" width="237px"/>

Runs a request in a collection and outputs the result. If a request that has a saved example was used, the result can be used as test data to run the flow and auto-fill **Select** blocks.

**Send** - When this receives any input from another block, the selected request is run.

**Environment** - Lets you select which Postman environment to use with your request.

**Body Type** - The format that your information uses (for example JSON or XML). **Auto** will try to auto-detect which schema is being used.

> **Variables** - If the request being run contains any variables in `{{variable}}` format, the will show up here. Variables can either be assigned a value in the block itself, have a value passed in using a **Select** block, or be already assigned in the Environment.

Reviewer: ```suggestion
**Variables** - If the request being run contains any variables in `{{variable}}` format, they will show up here. Variables can either be assigned a value in the block itself, have a value passed in using a **Select** block, or be already assigned in the Environment.
```

<!-- vale Postman.ComplexWords = NO -->
**Success** - Contains the response of a successful API request. If Postman tests are assigned to the request and all pass or, if there are no tests, a 2xx (successful) HTTP status code is received. The information returned by the request is then output to the next block.

**Failure** - Contains the response of an unsuccessful API request. If Postman tests are assigned to the request and all fail or, if there are no tests, a non 2xx (unsuccessful) HTTP status code is received. The information returned by the request is then output to the next block.
<!-- vale Postman.ComplexWords = YES -->

[Example](https://www.postman.com/postman/workspace/flows-snippets/flow/6267f9008752c1035922de48)

## Output blocks

### Log (console)

<img alt="Log (console) block" src="https://assets.postman.com/postman-labs-docs/all-blocks/log-block.png" width="78px"/>

Sends the information input to the block to the Postman console.

[Example](https://www.postman.com/postman/workspace/flows-snippets/flow/63b8b07ea365885b55bc51f9)

### Output

<img alt="Output block" src="https://assets.postman.com/postman-docs/v10/output-block-v10.jpg" width="249px"/>

> Displays the information input. The dropdown menu allows you to select from JSON, charts, tables, videos, images, and more. For more information and examples on the output block, see the [Creating charts, tables, and output](/docs/postman-flows/tutorials/creating-charts-tables-and-output/).

Reviewer: ```suggestion
Displays the information input. The dropdown menu allows you to select from JSON, charts, tables, videos, images, and more. For more information and examples on the output block, see [Creating charts, tables, and output](/docs/postman-flows/tutorials/creating-charts-tables-and-output/).
```
